ers/sql.go

```diff
@@ -184,23 +358,40 @@ func (rdr *SQLEventReader) readLoop(db *gorm.DB, sqlDB io.Closer) {
 						if tm.IsZero() {
 							continue
 						}
+						fltr[colName] = columns[i]
+						continue
 					case *time.Time:
 						if tm == nil || tm.IsZero() {
 							continue
 						}
+						fltr[colName] = columns[i]
+						continue
 					case nil:
 						continue
 					}
 					fltr[colName] = utils.IfaceAsString(columns[i])
 				}
 			}
-			if err = db.Table(rdr.tableName).Delete(nil, fltr).Error; err != nil { // to ensure we don't read it again
-				utils.Logger.Warning(
-					fmt.Sprintf("<%s> deleting message %s error: %s",
-						utils.ERs, utils.ToJSON(msg), err.Error()))
-				rdr.rdrErr <- err
-				rows.Close()
-				return
+			if rdr.Config().Opts.SQL.DeleteIndexedFields != nil {
```

Comment:
Here I would mix the if conditions in different way:
1. Only process the delete part of processedPath == *metaDelete.
2. Check first indexed fields.
3. If filterMp is still empty, add the fields out of colums.